<a href="https://colab.research.google.com/github/VarshaNair880/Statistics/blob/main/Lost_Key_problem.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Finding the right key


##Description:


You have lost the key to your house. Your houseowner gives you a bunch of 50 look-alike keys and asks you to find your key from the bunch. What is the probability that you get the right key in your fifth attempt?

##Computation:


The right key can be any of the 50 keys. The first case is when you get the right key on the first try. The second care is when the first key is the wrong key and the second key is the right key. Similarly, in the third case, the first two keys are wrong and the third is the right key. This sequence can be followed till 49 keys are wrong and the 50th key is the right key. Suppose 1 denotes the right key and 0 represents the wrong key, the sample space can be written as : 
##S = {1,01,001,0001,.........,0(49 times)1}

## This represents a Uniform distribution where the probability of getting the right key is 1/50 = 0.02.


#Simulation:


##Importing

Import package random to shuffle the list in random order.

In [ ]:
import random as rd 

##Defining a function

We define a function **find_right_key** with parameter **num_keys** to generate a set of keys which are shuffled for each trial.

In [ ]:
def find_right_key(num_keys):
    keys = list(range(num_keys))  #generating a list of keys
    rd.shuffle(keys)    #shuffling keys for different trials
    return keys[4] == 0  #checking if the 5th key in the list is the right key

##Finding the number of success

Specify the number of trials and initialise number of success to 0. Then repeat the experiment for the number of trials specified. 

In [ ]:
num_trials = 100000
num_success = 0

for i in range(num_trials):
    if find_right_key(50):
        num_success += 1


prob_success = num_success / num_trials
print("Probability of finding the right key in the fifth trial:", prob_success)

Probability of finding the right key in the fifth trial: 0.01996


#Analysis



The probability that you will get the right key in the fifth try is 1/50 which equals 0.02.
Monte Carlo Simulation gives value approximately 0.01996 which is close to 0.02. 
Hence, the manual computation matches the simulation!